x/evm/keeper/static_precompiles.go

```diff
@@ -0,0 +1,136 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper
+
+import (
+	"fmt"
+	"slices"
+
+	authzkeeper "github.com/cosmos/cosmos-sdk/x/authz/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	distributionkeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	channelkeeper "github.com/cosmos/ibc-go/v7/modules/core/04-channel/keeper"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	bankprecompile "github.com/evmos/evmos/v18/precompiles/bank"
+	"github.com/evmos/evmos/v18/precompiles/bech32"
+	distprecompile "github.com/evmos/evmos/v18/precompiles/distribution"
+	ics20precompile "github.com/evmos/evmos/v18/precompiles/ics20"
+	"github.com/evmos/evmos/v18/precompiles/p256"
+	stakingprecompile "github.com/evmos/evmos/v18/precompiles/staking"
+	vestingprecompile "github.com/evmos/evmos/v18/precompiles/vesting"
+	erc20Keeper "github.com/evmos/evmos/v18/x/erc20/keeper"
+	"github.com/evmos/evmos/v18/x/evm/types"
+	transferkeeper "github.com/evmos/evmos/v18/x/ibc/transfer/keeper"
+	stakingkeeper "github.com/evmos/evmos/v18/x/staking/keeper"
+	vestingkeeper "github.com/evmos/evmos/v18/x/vesting/keeper"
+	"golang.org/x/exp/maps"
+)
+
+// AvailableStaticPrecompiles returns the list of all available precompiled contracts.
+// NOTE: this should only be used during initialization of the Keeper.
+func AvailableStaticPrecompiles(
```

Comment:
I would suggest a renaming for this function because it is creating new instances of precompiles and it should be more explicit.

```suggestion
func NewAvailableStaticPrecompiles(
```